runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -924,7 +924,7 @@ J9::ARM64::TreeEvaluator::generateCheckForValueMonitorEnterOrExit(TR::Node *node
 
    generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmw, node, temp1Reg, classFlagsMemRef);
    loadConstant32(cg, node, classFlag, temp2Reg);
-   generateTrg1Src2Instruction(cg, TR::InstOpCode::andw, node, temp1Reg, temp1Reg, temp2Reg);
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::andsw, node, temp1Reg, temp1Reg, temp2Reg);
 
    bool generateOOLSection = helperCallLabel == NULL;
```

Comment:
The way the code reads it is possible that `helperCallLabel` will be NULL, meaning the conditional branch on L934 will target a NULL label.  Or am I reading/reasoning about this incorrectly?

Should this whole sequence of code be guarded with `generateOOLSection` ?

Reply:
At the very least L934 should be guarded with a `if (helperCallLabel)  { ... }`

Reply:
L.931 generates a new label and sets it to `helperCallLabel` when `helperCallLabel` is NULL at L.929.
L.934 does not require a guard.